src/python/zigopt/file/storage/s3.py

```diff
@@ -6,7 +6,6 @@
 from zigopt.file.storage.base import FileStorage
 from zigopt.json.builder import S3DownloadJsonBuilder, S3UploadJsonBuilder
 from zigopt.protobuf.gen.file.filedata_pb2 import S3StorageMethod
-from zigopt.protobuf.lib import CopyFrom
```

Comment:
gotcha yeah, i eventually pieced that all together when coming across the monkeypatching classes and utils-like functions. Odd / interesting; thanks for cleaning up!

Reply:
Oh, but: is there any reason we need `copy_protobuf(p)` to be distinct from `p.CopyFrom()`? if they serve (slightly?) different purposes that's fine, but if they're equivalent, then let's just stick with the builtin?

Reply:
an alternative is to use `copy.deepcopy()`, since the `__deepcopy__` method of the `Message` class is implemented. https://chromium.googlesource.com/external/github.com/google/protobuf/+/v3.2.0-rc.1/python/google/protobuf/message.py

Reply:
oh nice find! i see the reason for the 2 separate pieces; i'm happy with however you wanna go with it.